active/ItemsSourceView-Indexof/ItemsSourceView-IndexOf.md

```diff
@@ -0,0 +1,58 @@
+# Background
+<!-- Use this section to provide background context for the new API(s) 
+in this spec. -->
+The Xaml [ItemsRepeater](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ItemsRepeater) control displays a list of objects using a specified template (similar to a [ListView](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.ListView)). You provide the list of objects with the [ItemsRepeater.ItemsSource](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ItemsRepeater.ItemsSource) property. The list need only be an IIterable but can be an IVector for better perf. ItemsRepeater exposes a read-only [ItemsRepeater.ItemsSourceView](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ItemsRepeater.ItemsSourceView) property that provides a view on this list (an [ItemsSourceView](https://docs.microsoft.com/uwp/api/Microsoft.UI.Xaml.Controls.ItemsSourceView)), which has helpers like Count and GetAt over what might be just an IIterable.
```

Comment:
Should ItemsSourceView also implement IIterable, so you can
```cs
foreach (var item in ItemsRepeater.ItemsSource)
{
   ... something with item ...
}
```

Or is this not something people do frequently?

Reply:
That's a really good idea and also a really good question. There is definitely a point in having it be IIterable (IEnumerable in .Net). However if one would use the ItemsSourceView, they know its a IIterable, and thus could also do something like this:
```c#
foreach (var item in (IEnumerable)ItemsRepeater.ItemsSource)
{
   ... something with item ...
}
```

I am really not sure what would be best here. Something worth exploring might also be exposing the collection through a read only value, though that kinda defeats the point of ItemsSourceView in some areas.